src/main/java/com/mraof/minestuck/blockentity/machine/AnthvilBlockEntity.java

```diff
@@ -0,0 +1,259 @@
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * Mends an item at the cost of uranium power and grist. The grist used is the one with the highest value impact (weighted against build grist)
+ */
+public class AnthvilBlockEntity extends MachineProcessBlockEntity implements MenuProvider, UraniumPowered
+{
+	public static final String TITLE = "container.minestuck.anthvil";
+	public static final short MAX_FUEL = 128;
+	public static final short MEND_FUEL_COST = 5;
+	
+	private final ProgressTracker progressTracker = new ProgressTracker(ProgressTracker.RunType.ONCE, 0, this::setChanged, this::contentsValid);
+	private short fuel = 0;
+	
+	private final DataSlot fuelHolder = new DataSlot()
+	{
+		@Override
+		public int get()
+		{
+			return fuel;
+		}
+		
+		@Override
+		public void set(int value)
+		{
+			fuel = (short) value;
+		}
+	};
+	
+	
+	public AnthvilBlockEntity(BlockPos pos, BlockState state)
+	{
+		super(MSBlockEntityTypes.ANTHVIL.get(), pos, state);
+	}
+	
+	@Override
+	public Component getDisplayName()
+	{
+		return Component.translatable(TITLE);
+	}
+	
+	@Override
+	public void load(CompoundTag compound)
+	{
+		super.load(compound);
+		
+		this.progressTracker.load(compound);
+		
+		fuel = compound.getShort("fuel");
+	}
+	
+	@Override
+	public void saveAdditional(CompoundTag compound)
+	{
+		super.saveAdditional(compound);
+		
+		this.progressTracker.save(compound);
+		
+		compound.putShort("fuel", fuel);
+	}
+	
+	@Override
+	protected ItemStackHandler createItemHandler()
+	{
+		return new MachineProcessBlockEntity.CustomHandler(2, (index, stack) -> index == 0 || stack.is(ExtraForgeTags.Items.URANIUM_CHUNKS));
+	}
+	
+	@Override
+	protected void tick()
+	{
+		this.progressTracker.tick(this::processContents);
+	}
+	
+	private boolean contentsValid()
+	{
+		if(level.hasNeighborSignal(this.getBlockPos()))
+		{
+			return false;
+		}
+		
+		ItemStack fuel = itemHandler.getStackInSlot(1);
+		ItemStack input = itemHandler.getStackInSlot(0);
+		
+		boolean goodOnFuel = canBeRefueled() && fuel.is(ExtraForgeTags.Items.URANIUM_CHUNKS);
+		return goodOnFuel || !input.isEmpty();
+	}
+	
+	/**
+	 * Only handles refreshing uranium
+	 */
+	private void processContents()
+	{
+		if(canBeRefueled())
+		{
+			//checks for a uranium itemstack in the right item slot, increases the fuel value if some is found and then removes one count from the fuel stack
+			if(itemHandler.getStackInSlot(1).is(ExtraForgeTags.Items.URANIUM_CHUNKS))
+			{
+				//Refill fuel
+				addFuel((short) FUEL_INCREASE);
+				itemHandler.extractItem(1, 1, false);
+			}
+		}
+	}
+	
+	public static void attemptMend(AnthvilBlockEntity anthvil, ServerPlayer player)
+	{
+		Level level = anthvil.level;
+		ItemStack slotStack = anthvil.itemHandler.getStackInSlot(0);
+		GristCache playerCache = GristCache.get(player);
+		
+		ImmutableGristSet pickedGrist = mendingGrist(anthvil, playerCache, level, slotStack);
```

Comment:
```suggestion
		GristSet pickedGrist = mendingGrist(anthvil, playerCache, level, slotStack);
```
No reason for this to be immutable, leave the type as `GristSet` instead